main.go

```diff
@@ -732,9 +734,37 @@ func executeLinter(state *linterState, args []string) error {
 			state.Name)
 	}
 
-	processOutput(state, buf.Bytes())
+	cmdOutput, err := preprocessLinterPanic(state.Name, buf, err)
+	processOutput(state, cmdOutput)
 	debug("%s linter took %s", state.Name, time.Since(start))
-	return nil
+	return err
+}
+
+const panicExitStatus = 2
+
+func preprocessLinterPanic(name string, buf *bytes.Buffer, processError error) ([]byte, error) {
+	if exitStatus(processError) == panicExitStatus {
+		var err error
+		cmdOutput := new(bytes.Buffer)
+		routines, _ := stack.ParseDump(buf, cmdOutput)
```

Comment:
I disagree, the purpose here is to know whether it was a panic, if it couldn't reliably detect it was a panic, that doesn't matter. If anything, perhaps debug logging it, but I wouldn't bother myself.